app/controllers/application/cancelaccessrequest/CancelAccessRequestEndJourneyController.scala

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.application.cancelaccessrequest
+
+import com.google.inject.Inject
+import controllers.actions.{CancelAccessRequestDataRetrievalAction, DataRequiredAction, IdentifierAction}
+import controllers.helpers.ErrorResultBuilder
+import models.accessrequest.AccessRequest
+import models.application.Application
+import models.requests.DataRequest
+import models.user.UserModel
+import models.{CheckMode, UserAnswers}
+import pages.application.cancelaccessrequest.{CancelAccessRequestApplicationPage, CancelAccessRequestConfirmPage, CancelAccessRequestPendingPage, CancelAccessRequestSelectApiPage}
+import play.api.i18n.{I18nSupport, Messages}
+import play.api.mvc.*
+import repositories.CancelAccessRequestSessionRepository
+import services.ApiHubService
+import uk.gov.hmrc.http.{HeaderCarrier, UpstreamErrorResponse}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import views.html.application.cancelaccessrequest.CancelAccessRequestSuccessView
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class CancelAccessRequestEndJourneyController @Inject()(
+  override val controllerComponents: MessagesControllerComponents,
+  sessionRepository: CancelAccessRequestSessionRepository,
+  identify: IdentifierAction,
+  getData: CancelAccessRequestDataRetrievalAction,
+  requireData: DataRequiredAction,
+  cancelAccessRequestSuccessView: CancelAccessRequestSuccessView,
+  apiHubService: ApiHubService,
+  errorResultBuilder: ErrorResultBuilder
+)(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
+
+  import CancelAccessRequestEndJourneyController.*
+
+  def submitRequest(): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+      validate(request).fold(
+        call => Future.successful(Redirect(call)),
+        data => {
+          val accessRequests = getCancellableRequests(data)
+          Future.sequence(getCancellableRequests(data).map(accessRequest => apiHubService.cancelAccessRequest(accessRequest.id, request.user.email)))
+            .flatMap(_ => sessionRepository.clear(request.user.userId))
+            .flatMap(_ => Future.successful(Ok(cancelAccessRequestSuccessView(data.application, Some(request.user), accessRequests))))
```

Comment:
can this be a `map` ?
```suggestion
            .map(_ => Ok(cancelAccessRequestSuccessView(data.application, Some(request.user), accessRequests)))
```

Reply:
I don't think so, no. That changes the return type

[error]    |Found:    scala.concurrent.Future[scala.concurrent.Future[play.api.mvc.Result] |
[error]    |Required: scala.concurrent.Future[play.api.mvc.Result]

Reply:
oh I mean, we shouldn't return a `Future` inside the `map` but just a `Result`:
```
.flatMap(_ => Future(_ => Ok(...
```
to
```
.map(_ => Ok(...
```

Reply:
Ah ok